pylint/extensions/_check_docs_utils.py

```diff
@@ -230,7 +230,7 @@ def params_documented_elsewhere(self):
 
 class SphinxDocstring(Docstring):
     re_type = r"""
-        [~!]?                # Optional link style prefix
+        [~!.]?               # Optional link style prefix
```

Comment:
Do you want to allow the character `.` or any character? Because inside a regex `.` means any character, if you want to authorize `.` then you need to add `\.` .

Reply:
I was under the impression that this didn't apply within the square brackets:

```python
>>> import re
>>> for i in range(256):
...     c = chr(i)
...     if re.match(r'[~!.]', c) is not None:
...         print(f'i={i}, c={c}')
... 
i=33, c=!
i=46, c=.
i=126, c=~
```

Reply:
See: https://stackoverflow.com/a/19976308/1068170

> In a character class (square brackets) any character except `^`, `-`, `]` or `\` is a literal.

I'm happy to replace `.` with `\.` though if you think it'll catch people off guard.

Reply:
Ho, I didn't know that, my bad. Thank you for the explanation.

Reply:
@dhermes thanks for updates, TIL ))